### DISCUSSION TITLE
How to detect when tiptap editor is fully initialised (add loader)?

### DISCUSSION THREAD

**Discussion by piszczu4** [ueberdosis/tiptap#4762] [General Questions] [answered]
How can I add a `loading` indicator for my tiptap editor? Currently I'm using `editor = useEditor()` and `  if (!editor) return <Loader2 className='animate-spin' />;
` but animation does not work as constanly its rerender. On official tiptap website it work (https://tiptap.dev/docs/editor/api/extensions/table-of-contents). Any suggestions?

**Comment by abecirovic3** [comment, marked as answer]
Can the tiptap editor events be of any help to you?
https://tiptap.dev/docs/editor/api/events
You have the beforeCreate, create and destroy events.
Maybe you you could hook into them and set some flag accordingly.

**Reply by piszczu4** [reply]
Yes indeed I set a `loaded` state which is set to `false` by `onBeforeCreate` and then set to `true` on `onCreate` :) Thank you

**Reply by abecirovic3** [reply]
No worries, glad it worked.
